### PR TITLE
feat: android basic log in

### PR DESCRIPTION
**Pull request by ds300**
This PR resolves [CX-870]

## NOTE ⚡ 

There's a lot of import statement churn in this diff, sorry about that. We used VSCode's `organize imports` command to avoid lots of manual work, so it's all automated. Hope it's not too distracting. My recommendation would be to ignore the import statements and just check the main part of the files, which should mainly involve simple one-word substitutions.

### Description

Adds a temporary basic log-in flow for android

- Adds 'config' and 'auth' sections within the `GlobalStore`.
- Adds android-compatible stubs for our `NativeModules`. This prevents the app from crashing when importing files which use, e.g. `ARNotificationsManager`, which only exists on iOS for now.

image
image
image

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a [CHANGELOG.yml](/CHANGELOG.yml) entry or my changes do not require one.

[CX-870]: https://artsyproduct.atlassian.net/browse/CX-870

### CODE REVIEW

File: src/lib/AndroidApp.tsx [hunk 1]
```diff
@@ -1,4 +1,43 @@
...
+  )
+})
+
+export const App = () => (
+  <Theme>
+    <GlobalStoreProvider>
+      <Main />
+    </GlobalStoreProvider>
+  </Theme>
+)
```
Comment (brainbicycle) [inline src/lib/AndroidApp.tsx:43]: This is a very exciting file 🎉

File: src/lib/NativeModules/ArtsyNativeModules.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      updateShouldHideBackButton: noop("updateShouldHideBackButton"),
+      presentAugmentedRealityVIR: noop("presentAugmentedRealityVIR"),
+      presentEmailComposer: noop("presentEmailComposer"),
+      popStack: noop("popStack"),
+      popToRootAndScrollToTop: noop("popToRootAndScrollToTop"),
+      popToRootOrScrollToTop: noop("popToRootOrScrollToTop"),
+      presentModal: noop("presentModal"),
+    },
+  },
+})
```
Comment (brainbicycle) [inline src/lib/NativeModules/ArtsyNativeModules.tsx:91]: 👍 Nice

File: src/lib/store/AuthModel.ts [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+  signOut: action((state) => {
+    Object.assign(state, {
+      userID: null,
+      userAccessToken: null,
+      userAccessTokenExpiresIn: null,
+      xAppToken: null,
+      xApptokenExpiresIn: null,
+    })
+  }),
+}
```
Comment (brainbicycle) [inline src/lib/store/AuthModel.ts:136]: 🔒 Nice this is going to be much easier to understand

File: src/lib/store/__tests__/AuthModel-tests.ts [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+      const token = await GlobalStore.actions.auth.getXAppToken()
+
+      expect(token).toBe("my-special-token")
+      expect(mockFetch).toHaveBeenCalledTimes(1)
+      expect(mockFetch.mock.calls[0][0]).toMatchInlineSnapshot(
+        `"https://stagingapi.artsy.net/api/v1/xapp_token?client_id=artsy_api_client_key&client_secret=artsy_api_client_secret"`
+      )
+    })
+
+    it("are saved to the store", async () => {
```
Comment (brainbicycle) [inline src/lib/store/__tests__/AuthModel-tests.ts:37]: **nitpick (non-blocking):**

```suggestion
    it("saves token values to the store", async () => {
```

File: src/lib/store/__tests__/AuthModel-tests.ts [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+    it("makes a request to gravity's /user endpoint", async () => {
+      mockFetchResponseOnce({ status: 200 })
+      await GlobalStore.actions.auth.userExists({ email: "user@example.com" })
+      expect(mockFetch).toHaveBeenCalledTimes(1)
+      expect(mockFetch.mock.calls[0][0]).toMatchInlineSnapshot(
+        `"https://stagingapi.artsy.net/api/v1/user?email=user%40example.com"`
+      )
+    })
+
+    it("returns true if repsonse is 200", async () => {
```
Comment (brainbicycle) [inline src/lib/store/__tests__/AuthModel-tests.ts:82]:
```suggestion
    it("returns true if response is 200", async () => {
```

File: src/lib/store/__tests__/AuthModel-tests.ts [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+      mockFetchResponseOnce({ status: 200 })
+      const result = await GlobalStore.actions.auth.userExists({ email: "user@example.com" })
+
+      expect(result).toBe(true)
+    })
+
+    it("returns false if repsonse is 404", async () => {
```
Comment (brainbicycle) [inline src/lib/store/__tests__/AuthModel-tests.ts:89]:
```suggestion
    it("returns false if response is 404", async () => {
```

File: src/lib/store/__tests__/AuthModel-tests.ts [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+        },
+        201
+      )
+      mockFetchJsonOnce({
+        id: "my-user-id",
+      })
+
+      const result = await GlobalStore.actions.auth.signIn({
+        email: "user@example.com",
+        password: "hunter2",
```
Comment (brainbicycle) [inline src/lib/store/__tests__/AuthModel-tests.ts:134]: 😆

File: src/lib/store/__tests__/AuthModel-tests.ts [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+    it("returns false if the token creation fails", async () => {
+      mockFetchJsonOnce({ error: "bad times" }, 500)
+      const result = await GlobalStore.actions.auth.signIn({
+        email: "user@example.com",
+        password: "hunter2",
+      })
+      expect(result).toBe(false)
+    })
+  })
+})
```
Comment (brainbicycle) [inline src/lib/store/__tests__/AuthModel-tests.ts:183]: Nice tests!

**Review by brainbicycle — APPROVED** [review]
Awesome work! Merge when ready

File: src/lib/LogIn/LogInEnterPassword.tsx [hunk 9]
```diff
@@ -0,0 +1,51 @@
+import { Input } from "lib/Components/Input/Input"
+import { Stack } from "lib/Components/Stack"
+import { GlobalStore } from "lib/store/GlobalStore"
+import { Button, Flex, Separator, Text } from "palette"
+import React from "react"
+import { SafeAreaView } from "react-native"
+import { ScrollView } from "react-native-gesture-handler"
```
Comment (pvinis) [inline src/lib/LogIn/LogInEnterPassword.tsx:7]: Just curious @ds300, is this on purpose or accidental? Does this have anything better/different than the react-native one?

Comment (ds300) [reply src/lib/LogIn/LogInEnterPassword.tsx:7]: this was an accident, but it's no biggie.

The only difference is that this one is compatible with react-native-reanimated

Comment (pvinis) [reply src/lib/LogIn/LogInEnterPassword.tsx:7]: no biggie indeed. I was only worried if I'm using the wrong one 😅. All good!

**Pull request merged**